src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,96 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private Object[] array = new Object[10];
+    private int size = 1;
+
     @Override
     public void add(T value) {
-
+        if (size == array.length) {
+            Object[] newArray = new Object[array.length + array.length / 2];
+            System.arraycopy(array, 0, newArray, 0, array.length);
+            array = newArray;
+        }
+        array[size - 1] = value;
+        size++;
     }
 
     @Override
-    public void add(T value, int index) {
-
+    public void add(T value, int index) throws ArrayListIndexOutOfBoundsException {
+        if (index > size() || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index out of bounds");
+        }
+        if (size >= array.length) {
+            Object[] newArray = new Object[array.length + array.length / 2];
+            System.arraycopy(array, 0, newArray, 0, array.length);
+            array = newArray;
+        }
+        System.arraycopy(array, index, array, index + 1, size() - index);
+        array[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        int summarySize = list.size() + size();
+        final int tempSize = summarySize;
+        summarySize = summarySize + summarySize / 2;
+        Object[] newArray = new Object[summarySize];
+        System.arraycopy(array, 0, newArray, 0, array.length);
+        array = newArray;
+        for (int i = 0; i < list.size(); i++) {
+            array[i + size - 1] = list.get(i);
+        }
+        size = tempSize + 1;
```

Comment:
The size update `size = tempSize + 1;` is incorrect. It should be `size = tempSize;` since `tempSize` already accounts for the total number of elements after adding the list.